### PR TITLE
Cloudsdk 27

### PR DESCRIPTION
**Pull request by Agreon**

### CODE REVIEW

File: lib/DcHttpClient.js [hunk 1]
```diff
@@ -117,6 +117,38 @@ class DcHttpClient {
...
   }
+
+  /**
+   *
+   * @param {Object} config
+   * @param {String} applicationId
+   * @param {Function} cb
+   */
+  generateExtensionConfig (config, applicationId, cb) {
+    console.log(applicationId, config)
```
Comment (xFragger) [inline lib/DcHttpClient.js]: console.log

File: lib/actions/BackendAction.js [hunk 2]
```diff
@@ -83,6 +91,30 @@ class BackendAction {
...
+    }, (err, extConfig) => {
+      if (err) {
+        logger.error(err.message)
+        return cb(err)
+      }
+      if (!extConfig.backend) {
+        logger.error('Backend-part is missing in generated config')
+        return cb(new Error('Backend-part is missing in generated config'))
+      }
+      fsEx.outputFileSync(path.join('', config.path, 'extension', 'config.json'), JSON.stringify(extConfig.backend, null, 2), 'UTF-8')
```
Comment (xFragger) [inline lib/actions/BackendAction.js]: use fs-extra's json output method

File: lib/actions/BackendAction.js [hunk 3]
```diff
@@ -83,6 +91,30 @@ class BackendAction {
...
+      this.dcClient.generateExtensionConfig(config.file, AppSettings.getInstance().getId(), (err, extConfig) => {
+        if (err) return acb(new Error(`Could not generate Config for '${config.file.id}'`))
+        return acb(null, extConfig)
+      })
+    }, (err, extConfig) => {
+      if (err) {
+        logger.error(err.message)
+        return cb(err)
+      }
+      if (!extConfig.backend) {
```
Comment (pvomhoff) [inline lib/actions/BackendAction.js]: This should not lead to an error.
It should just do nothing

File: lib/DcHttpClient.js [hunk 4]
```diff
@@ -47,7 +48,8 @@ class DcHttpClient {
     }
 
     request(opts, (err, res, body) => {
-      if (err || res.statusCode !== 200) return cb(err || new Error(body ? body.message : `could not get ${infoType}`))
+      if (err) return cb(err)
+      if (res.statusCode !== 200) return cb(err || new Error(body ? body.message : `could not get ${infoType}`))
```
Comment (daddy-cool) [inline lib/DcHttpClient.js]: remove err from the cb as it will always be falsy

File: lib/DcHttpClient.js [hunk 5]
```diff
@@ -21,7 +21,8 @@ class DcHttpClient {
     }
 
     request(opts, (err, res, body) => {
-      if (err || res.statusCode !== 200) return cb(err || new Error(body ? body.message : 'Login failed'))
+      if (err) return cb(err)
+      if (res.statusCode !== 200) return cb(err || new Error(body ? body.message : 'Login failed'))
```
Comment (daddy-cool) [inline lib/DcHttpClient.js]: remove err from the cb as it will always be falsy

File: lib/actions/BackendAction.js [hunk 6]
```diff
@@ -89,6 +97,31 @@ class BackendAction {
       cb()
     })
   }
+
+  _extensionChanged (config, cb = () => {}) {
+    async.retry({
+      times: 5,
+      interval: (count) => { return 100 * Math.pow(2, count) }
```
Comment (daddy-cool) [inline lib/actions/BackendAction.js]: remove curly braces and keyword "return"

File: lib/actions/BackendAction.js [hunk 7]
```diff
@@ -28,17 +29,20 @@ class BackendAction {
...
 
     process.on('SIGINT', () => {
       this.pipelineWatcher.stop(() => {
-        this.backendProcess.extensionWatcher.close((err) => {
-          if (err) this.log.error(err)
-          this.backendProcess.disconnect((err2) => {
-            if (err2) this.log.error(err2)
-            logger.info('SDK connection closed')
-            process.exit((err || err2) ? 1 : 0)
+        this.extensionConfigWatcher.stop(() => {
```
Comment (daddy-cool) [inline lib/actions/BackendAction.js]: use async parellel

File: lib/actions/BackendAction.js [hunk 8]
```diff
@@ -28,17 +29,20 @@ class BackendAction {
...
+          this.backendProcess.extensionWatcher.close((err) => {
+            if (err) this.log.error(err)
+            this.backendProcess.disconnect((err2) => {
+              if (err2) this.log.error(err2)
+              logger.info('SDK connection closed')
```
Comment (daddy-cool) [inline lib/actions/BackendAction.js]: only log when there was no error

**Pull request merged**